compiler_gym/datasets/files_dataset.py

```diff
@@ -0,0 +1,156 @@
+                    filename = filename[: -len(self.benchmark_file_suffix)]
+                # Use os.path.join() rather than simple '/' concaentation as
+                # reldir may be empty.
+                yield os.path.join(self.name, reldir, filename)
+
+    @property
+    def _benchmark_uris(self) -> List[str]:
+        return list(self._benchmark_uris_iter)
+
+    def benchmark_uris(self) -> Iterable[str]:
+        if self._memoized_uris:
+            yield from self._memoized_uris
+        elif self.memoize_uris:
+            self._memoized_uris = self._benchmark_uris
+            yield from self._memoized_uris
+        else:
+            yield from self._benchmark_uris_iter
+
+    def benchmark(self, uri: Optional[str] = None) -> Benchmark:
+        self.install()
+        if uri is None or len(uri) <= len(self.name) + 1:
+            if not self.n:
+                raise ValueError("No benchmarks")
+            return self.get_benchmark_by_index(self.random.integers(self.n))
+
+        relpath = f"{uri[len(self.name) + 1:]}{self.benchmark_file_suffix}"
+        abspath = self.dataset_root / relpath
+        if not abspath.is_file():
+            raise LookupError(f"Benchmark not found: {uri} (file not found: {abspath})")
+        return self.benchmark_class.from_file(uri, abspath)
+
+    def get_benchmark_by_index(self, n: int) -> Benchmark:
+        """Look a benchmark using a numeric index into the list of URIs."""
+        # If we have memoized the benchmark IDs then just index into the list.
+        # Otherwise we will scan through the directory hierarchy.
+        if self.memoize_uris:
+            if not self._memoized_uris:
+                self._memoized_uris = self._benchmark_uris
+            return self.benchmark(self._memoized_uris[n])
+
+        i = 0
+        for root, dirs, files in os.walk(self.dataset_root):
+            reldir = root[len(str(self.dataset_root)) + 1 :]
+
+            # Filter only the files that match the target file suffix.
+            valid_files = [f for f in files if f.endswith(self.benchmark_file_suffix)]
+
+            if i + len(valid_files) < n:
+                # There aren't enough files in this directory to bring us up to
+                # the target file index, so skip the whole lot.
+                i += len(valid_files)
+                dirs.sort()
+            else:
+                # Iterate through the files in the directory in order until we
+                # reach the target index.
+                for filename in sorted(valid_files):
+                    if i == n:
+                        name_stem = filename[: -len(self.benchmark_file_suffix)]
+                        # Use os.path.join() rather than simple '/' concaentation as
+                        # reldir may be empty.
+                        uri = os.path.join(self.name, reldir, name_stem)
+                        return self.benchmark_class.from_file(uri, f"{root}/{filename}")
+                    i += 1
```

Comment:
I am bit confused here. Doesn't *files* contain all files in the directory?

Reply:
You're right, this loop wasn't required. Removed